src/usr/share/gpgmailer/gpgkeyverifier.py

```diff
@@ -93,68 +94,55 @@ def get_expiration_warning_message(self, loop_current_time):
     #   with a receiver.
     #
     # config: The config dictionary read from the program configuration file.
-    def _initialize_email_dicts(self, config):
+    def _initialize_key_dicts(self, config):
 
-        recipient_fingerprints = []
+        # Build a dict of fingerprints from the keyring that includes associated email addresses,
+        # whether the key is the gpgmailer configured sender, and whether any expiry emails have
+        # been sent for this key.
+        fp_to_key_dict = self.gpgkeyring.fingerprint_to_key_dict
+        for fingerprint in fp_to_key_dict:
 
-        # Record e-mail information for all the recipients.
-        for recipient in config['recipients']:
-            # TODO: Eventually, handle multiple keys for one address.
-            if recipient['email'] in self.all_recipient_emails:
-                raise RecipientEmailCollision('Email %s is already configured.' % recipient['email'])
-
-            email_dict = { 'fingerprint': recipient['fingerprint'],
+            key_dict = {
+                'fingerprint': fp_to_key_dict[fingerprint]['fingerprint'],
+                'email_list': fp_to_key_dict[fingerprint]['emails'],
                 'expired_email_sent': False,
-                'expiring_soon_email_sent': False,
-                'is_sender': False,
-                'is_recipient': True }
-
-            recipient_fingerprints.append(recipient['fingerprint'])
-            self.email_dicts[recipient['email']] = email_dict
-            self.all_recipient_emails.append(recipient['email'])
+                'expiring_soon_email_sent': False
+            }
 
-        # The sender might also be a receipient.
-        self.sender_email = config['sender']['email']
-        if self.sender_email in self.email_dicts.keys():
-            # The sender is also a recipient.
-            self.email_dicts[self.sender_email]['is_sender'] = True
+            # Mark the gpgmailer configured sender key.
+            if fp_to_key_dict[fingerprint]['fingerprint'] == config['sender']['fingerprint']:
+                self.sender_key = key_dict
```

Comment:
It would probably be faster to lookup the sender key by fingerprint after the loop is finished.

Reply:
Done